### ISSUE TITLE
How to individually train the predictor?

### ISSUE THREAD

**Issue by YYX666660** [Crystalsound/FRN#4]
Hello! 
Since I did not use your dataset, the effect of using my own dataset and the predictor you provided was not very good. Therefore, I would like to use my dataset to train the predictor separately first, and then finetune it on my dataset. How should I do this work?
Thank you so much!

**Comment by vietanh125** [comment]
Training the predictor is quite straightforward, you will train it with next stft frame prediction. Firstly, STFT-transform the input signal to obtain a sequence of stft frames, this will be the output. Then right-shift the frame sequence, trim the last frame and left-pad to get the input. Since the predictor is a unidirectional LSTM, it's autoregressive itself, so you just need to feed in the entire input sequence, obtain the predicted sequence and calculate the loss.

Hope it helps.
